tests/v3_api/test_secrets.py

```diff
@@ -0,0 +1,331 @@
+    print result
+
+    print "Verify that the secret does not exist " \
+          "and the error code returned is non zero "
+    if result != 0:
+        assert True
+
+
+def test_delete_secret_single_ns():
+
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    name = random_test_name("default")
+    keyvaluepair = {"testadelete": "valuedelete"}
+
+    secret = p_client.create_namespaced_secret(name=name,
+                                               namespaceId=ns['name'],
+                                               stringData=keyvaluepair)
+    print secret
+    assert secret['baseType'] == "namespacedSecret"
+    secretdata = secret['data']
+
+    for k, v in secretdata.items():
+        key = k
+        vdecoded = base64.b64decode(v)
+
+    assert key == keyvaluepair.keys()[0]
+    assert vdecoded == keyvaluepair.values()[0]
+
+    print "Delete Secret"
+    p_client.delete(secret)
+
+    # Sleep to allow for the secret to be deleted
+    time.sleep(5)
+    print "Secret list after deleting one secret"
+    secretdict = p_client.list_namespaced_secret()
+    print secretdict
+    print secretdict.get('data')
+    if len(secretdict.get('data')) == 0:
+        assert True
+
+    # Verify secret is deleted by "kubectl get secret" command
+    command = " get secret " + secret['name'] + " --namespace=" + ns.name
+    print "Command to obtain the secret"
+    print command
+    result = execute_kubectl_cmd(command, json_out=False, stderr=True)
+    print result
+
+    print "Verify that the secret does not exist and the error " \
+          "code returned is non zero "
+    if result != 0:
+        assert True
+
+
+def test_edit_secret_all_ns():
+
+    p_client = namespace["p_client"]
+    name = random_test_name("default")
+    keyvaluepair = {"testall": "valueall"}
+
+    secret = p_client.create_secret(name=name, namespaceId='NULL',
+                                    stringData=keyvaluepair)
+    print secret
+    assert secret['baseType'] == "secret"
+    secretdata = secret['data']
+
+    for k, v in secretdata.items():
+        key = k
+        vdecoded = base64.b64decode(v)
+
+    assert key == keyvaluepair.keys()[0]
+    assert vdecoded == keyvaluepair.values()[0]
+
+    newkeyvaluepair = {"testallnew": "valueallnew"}
+    updated_dict = {"testall": "valueall", "testallnew": "valueallnew"}
+    updated_secret = p_client.update(secret, name=name, namespaceId='NULL',
+                                     stringData=newkeyvaluepair)
+
+    assert updated_secret['baseType'] == "secret"
+    updatedsecretdata = updated_secret['data']
+
+    print updatedsecretdata
+
+    finalkeyvaluepair = {}
+    for k, v in updatedsecretdata.items():
+        keytest = k
+        vdecoded = base64.b64decode(v)
+        finalkeyvaluepair.update({keytest: vdecoded})
+
+    assert len(updatedsecretdata.items()) == 2
+    print finalkeyvaluepair
+    assert cmp(updated_dict, finalkeyvaluepair) == 0
```

Comment:
Validate that the updated secret values are present in the new workloads that get deployed with it . Or we can also try to upgrade an existing workload that was deployed before secret update and make sure that new values set as secret upgrade is available for the upgraded pods.

Reply:
Can we also check this with multiple  namespaces in projects?

Reply:
Created new workloads with the updated secrets. Also added check by created a new namespace